File: app/src/main/java/com/webkul/mobikul/odoo/firebase/FirebaseAnalyticsImpl.java
package com.webkul.mobikul.odoo.firebase;

import android.content.Context;
import android.os.Bundle;

import com.google.firebase.analytics.FirebaseAnalytics;


/**

 * Webkul Software.

 * @package Mobikul App

 * @Category Mobikul

 * @author Webkul <support@webkul.com>

 * @Copyright (c) Webkul Software Private Limited (https://webkul.com)

 * @license https://store.webkul.com/license.html ASL Licence

 * @link https://store.webkul.com/license.html

 */

public class FirebaseAnalyticsImpl {



    private static FirebaseAnalytics mFirebaseAnalytics;

    public FirebaseAnalyticsImpl(Context context) {
        // Obtain the FirebaseAnalytics instance.
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
    }

    public static void logViewItem(Context context, String productId, String productName) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        try {
            bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, Integer.parseInt(productId));
        } catch (NumberFormatException e) {
            bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, 0);
        }
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, productName);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.VIEW_ITEM, bundle);
    }
    public static void logEcommercePurchase(Context context, String productName) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, productName);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.VIEW_ITEM, bundle);
    }

    public static void logAppOpenEvent(Context context) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.APP_OPEN, null);
    }

    public static void logAddToWishlistEvent(Context context, String productId, String productName) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        try {
            bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, Integer.parseInt(productId));
        } catch (NumberFormatException e) {
            bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, 0);
        }
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, productName);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.ADD_TO_WISHLIST, bundle);
    }


    public static void logAddToCartEvent(Context context, String productId, String productName) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        try {
            bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, Integer.parseInt(productId));
        } catch (NumberFormatException e) {
            bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, 0);
        }
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, productName);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.ADD_TO_CART, bundle);

    }

    public static void logBeginCheckoutEvent(Context context) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.BEGIN_CHECKOUT, null);
    }


    public static void logLoginEvent(Context context, String customerId, String customerName) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, Integer.parseInt(customerId));
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, customerName);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.LOGIN, bundle);
    }


    public static void logSearchEvent(Context context, String searchQuery) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, searchQuery);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.SEARCH, bundle);
    }

    @SuppressWarnings("unused")
    public static void logShareEvent(Context context, String productId, String productName) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, Integer.parseInt(productId));
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, productName);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.SHARE, bundle);
    }

    public static void logSignUpEvent(Context context, String customerId, String customerName) {
        if (mFirebaseAnalytics == null) {
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, Integer.parseInt(customerId));
        bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, customerName);
        mFirebaseAnalytics.logEvent(FirebaseAnalytics.Event.SIGN_UP, bundle);
    }

    public static void logPostalCodeUnavailable(Context context, String state,String district,String subdistrict,String village){
        if(mFirebaseAnalytics == null){
            mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
        }
        Bundle bundle = new Bundle();
        bundle.putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_STATE,state);
        bundle.putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_DISTRICT,district);
        bundle.putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_SUB_DISTRICT,subdistrict);
        bundle.putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_VILLAGE,village);
        mFirebaseAnalytics.logEvent(CustomFirebaseAnalyticsEvent.NON_SERVICEABLE_LOCATION_EVENT,bundle);
    }
}

File: app/src/main/java/com/webkul/mobikul/odoo/core/utils/FirebaseAnalyticsImpl.kt
package com.webkul.mobikul.odoo.core.utils

import android.content.Context
import android.os.Bundle
import com.google.firebase.analytics.FirebaseAnalytics
import com.webkul.mobikul.odoo.firebase.CustomFirebaseAnalyticsEvent
import javax.inject.Inject

class FirebaseAnalyticsImpl @Inject constructor(private val context: Context) {

    fun logViewItem(productId: String, productName: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.VIEW_ITEM, Bundle().apply {
            try {
                putInt(FirebaseAnalytics.Param.ITEM_ID, productId.toInt())
            } catch (e: NumberFormatException) {
                putInt(FirebaseAnalytics.Param.ITEM_ID, 0)
            }
            putString(FirebaseAnalytics.Param.ITEM_NAME, productName)
        })
    }

    fun logEcommercePurchase(productName: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.VIEW_ITEM, Bundle().apply {
            putString(FirebaseAnalytics.Param.ITEM_NAME, productName)
        })
    }

    fun logAppOpenEvent() = FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.APP_OPEN, null)

    fun logAddToWishlistEvent(productId: String, productName: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.ADD_TO_WISHLIST, Bundle().apply {
            try {
                putInt(FirebaseAnalytics.Param.ITEM_ID, productId.toInt())
            } catch (e: NumberFormatException) {
                putInt(FirebaseAnalytics.Param.ITEM_ID, 0)
            }
            putString(FirebaseAnalytics.Param.ITEM_NAME, productName ?: "")
        })
    }

    fun logAddToCartEvent(productId: String, productName: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.ADD_TO_CART, Bundle().apply {
            try {
                putInt(FirebaseAnalytics.Param.ITEM_ID, productId.toInt())
            } catch (e: NumberFormatException) {
                putInt(FirebaseAnalytics.Param.ITEM_ID, 0)
            }
            putString(FirebaseAnalytics.Param.ITEM_NAME, productName ?: "")
        })
    }

    fun logBeginCheckoutEvent() = FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.BEGIN_CHECKOUT, null)

    fun logLoginEvent(customerId: String, customerName: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.LOGIN, Bundle().apply {
            putInt(FirebaseAnalytics.Param.ITEM_ID, customerId.toInt())
            putString(FirebaseAnalytics.Param.ITEM_NAME, customerName ?: "")
        })
    }

    fun logSearchEvent(searchQuery: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.SEARCH, Bundle().apply {
            putString(FirebaseAnalytics.Param.ITEM_NAME, searchQuery ?: "")
        })
    }

    fun logShareEvent(productId: String, productName: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.SHARE, Bundle().apply {
            putInt(FirebaseAnalytics.Param.ITEM_ID, productId.toInt())
            putString(FirebaseAnalytics.Param.ITEM_NAME, productName ?: "")
        })
    }

    fun logSignUpEvent(customerId: String, customerName: String?) {
        FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.SIGN_UP, Bundle().apply {
            putInt(FirebaseAnalytics.Param.ITEM_ID, customerId.toInt())
            putString(FirebaseAnalytics.Param.ITEM_NAME, customerName ?: "")
        })
    }

    fun logPostalCodeUnavailable(
            state: String?,
            district: String?,
            subdistrict: String?,
            village: String?
    ) {
        FirebaseAnalytics.getInstance(context).logEvent(
                CustomFirebaseAnalyticsEvent.NON_SERVICEABLE_LOCATION_EVENT,
                Bundle().apply {
                    putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_STATE, state ?: "")
                    putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_DISTRICT, district ?: "")
                    putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_SUB_DISTRICT, subdistrict
                            ?: "")
                    putString(CustomFirebaseAnalyticsEvent.UNSERVICEABLE_VILLAGE, village ?: "")
                })
    }
}